test/spec/userSync_spec.js

```diff
@@ -0,0 +1,162 @@
+import { expect } from 'chai';
+// Use require since we need to be able to write to these vars
+const utils = require('../../src/utils');
+let { userSync, newUserSync } = require('../../src/userSync');
+
+describe('user sync', () => {
+  let triggerPixelStub;
+  let logWarnStub;
+  let timeoutStub;
+  let shuffleStub;
+  let insertElementStub;
+  let getUniqueIdentifierStrStub;
+  let idPrefix = 'test-generated-id-';
+  let lastId = 0;
+
+  beforeEach(() => {
+    triggerPixelStub = sinon.stub(utils, 'triggerPixel');
+    logWarnStub = sinon.stub(utils, 'logWarn');
+    shuffleStub = sinon.stub(utils, 'shuffle', (array) => array.reverse());
+    getUniqueIdentifierStrStub = sinon.stub(utils, 'getUniqueIdentifierStr', () => idPrefix + (lastId += 1));
+    timeoutStub = sinon.stub(window, 'setTimeout', (callbackFunc) => { callbackFunc(); });
+  });
+
+  afterEach(() => {
+    triggerPixelStub.restore();
+    logWarnStub.restore();
+    shuffleStub.restore();
+    getUniqueIdentifierStrStub.restore();
+    timeoutStub.restore();
+    userSync = newUserSync();
+  });
+
+  it('should register and fire a pixel URL', () => {
+    userSync.registerSync('image', 'testBidder', 'http://example.com');
+    userSync.syncUsers();
```

Comment:
are there any tests that verify that `syncUsers()` is called when an auction finishes, rather than calling it directly?